libs/sdk-ui-pivot/src/impl/agGridColumnSizing.ts

```diff
@@ -658,3 +676,239 @@ export const resizeWeakMeasureColumns = (
+
+            if (config.cache) {
+                possibleMaxWidth = getMaxWidthCached(context, textForCalculation, maxWidth, config.cache);
+            } else {
+                possibleMaxWidth = getMaxWidth(context, textForCalculation, false, maxWidth);
+            }
+
+            if (possibleMaxWidth) {
+                maxWidths.set(colDef.field, possibleMaxWidth);
+            }
+        }
+    });
+};
+
+export const getUpdatedColumnDefs = (
+    columns: Column[],
+    maxWidths: Map<string, number>,
+    padding: number,
+): ColDef[] => {
+    return columns.map((column: Column) => {
+        const colDef: ColDef = column.getColDef();
+        if (colDef.field) {
+            const maxWidth = maxWidths.get(colDef.field);
+            const newWidth = maxWidth ? Math.ceil(maxWidth + padding) : 0;
+            return {
+                ...colDef,
+                width: Math.min(Math.max(MIN_WIDTH, newWidth), AUTO_SIZED_MAX_WIDTH),
+            };
+        }
+        return colDef;
+    });
+};
+
+const calculateColumnWidths = (config: CalculateColumnWidthsConfig) => {
+    const { context } = config;
+    const maxWidths = new Map<string, number>();
+
+    if (config.measureHeaders && context) {
+        context.font = config.headerFont;
+
+        config.columns.forEach((column: Column) => {
+            const colDef: ColDef = column.getColDef();
+            const maxWidth = colDef.field ? maxWidths.get(colDef.field) : undefined;
+            const possibleMaxWidth = getMaxWidth(context, colDef.headerName, !!colDef.sort, maxWidth);
+
+            if (colDef.field && possibleMaxWidth) {
+                maxWidths.set(colDef.field, possibleMaxWidth);
+            }
+        });
+    }
+
+    config.rowData.forEach((row: IGridRow) => {
+        if (context) {
+            context.font = isSomeTotal(row.type) ? config.subtotalFont : config.rowFont;
+            collectWidths(config, row, maxWidths);
+        }
+    });
+
+    config.totalData.forEach((row: IGridRow) => {
+        if (context) {
+            context.font = config.totalFont;
+            collectWidths(config, row, maxWidths);
+        }
+    });
+
+    return getUpdatedColumnDefs(config.columns, maxWidths, config.padding);
+};
+
+const getDisplayedRowData = (gridApi: GridApi): IGridRow[] => {
+    const rowCount = gridApi.getDisplayedRowCount();
+    const rowData: IGridRow[] = [];
+    for (let index = 0; index < rowCount; index++) {
+        const item: IGridRow = gridApi.getDisplayedRowAtIndex(index).data;
+        rowData.push(item);
+    }
+    return rowData;
+};
+
+const getDisplayedTotalData = (gridApi: GridApi): IGridRow[] => {
+    const totalCount = gridApi.getPinnedBottomRowCount();
+    const totalData: IGridRow[] = [];
+    for (let index = 0; index < totalCount; index++) {
+        const item: IGridRow = gridApi.getPinnedBottomRow(index).data;
+        totalData.push(item);
+    }
+    return totalData;
+};
+
+export const autoresizeAllColumns = (
```

Comment:
need documentation on this entry point function. describe in high level what the algorithm does, link over to the page outlining this approach.

Reply:
Done.